tests/test_middleware.py

```diff
@@ -249,6 +267,48 @@ def allow_all(sender, request, **kwargs):
 
             assert allow_all.calls == 1
 
+    @override_settings(CORS_ORIGIN_WHITELIST=['example.com'])
+    @prepend_middleware('tests.test_middleware.ShortCircuitMiddleware')
+    def test_process_response_with_short_circuit(self):
+        """
+        Test a scenario when a middleware that returns a response is run before
+        the ``CorsMiddleware``. In this case
+        ``CorsMiddleware.process_response()`` should ignore the request if
+        MIDDLEWARE setting is used (new mechanism in Django 1.10+) and process
+        the request and add CORS response headers if MIDDLEWARE_CLASSES is
+        used in a backward compatible fashion with django-cors-headers pre 1.3.0.
+        """
+        resp = self.client.get('/', HTTP_ORIGIN='http://example.com')
+        if django.VERSION >= (1, 10):
+            assert ACCESS_CONTROL_ALLOW_ORIGIN not in resp
+        else:
+            assert ACCESS_CONTROL_ALLOW_ORIGIN in resp
+
+    @override_settings(CORS_URLS_REGEX='^/abc/.*$')
+    @override_settings(CORS_ORIGIN_WHITELIST=['example.com'])
+    @prepend_middleware('tests.test_middleware.ShortCircuitMiddleware')
+    def test_process_response_with_short_circuit_no_origin(self):
+        resp = self.client.get('/def', HTTP_ORIGIN='http://example.com')
+        assert ACCESS_CONTROL_ALLOW_ORIGIN not in resp
+
+    @override_settings(CORS_URLS_REGEX='^/abc/.*$')
+    @override_settings(CORS_ORIGIN_WHITELIST=['example.com'])
+    def test_process_response_no_origin_not_enabled(self):
+        resp = self.client.get('/def', HTTP_ORIGIN='http://example.com')
+        assert ACCESS_CONTROL_ALLOW_ORIGIN not in resp
+
+    @override_settings(CORS_URLS_REGEX='^/abc/.*$')
+    @override_settings(CORS_ORIGIN_WHITELIST=['example.com'])
+    def test_process_response_no_origin_not_enabled_new_middleware(self):
+        # This tests the same thing as
```

Comment:
I didn't like this test which was inventing a scenario that's literally impossible on Django 1.10+ just to bump the coverage.

I replaced it with a test that calls a malicious view that removes `_cors_enabled` from the request, which is a somewhat more realistic test since it's actually possible.